eth/tracers/api_blocktrace.go

```diff
@@ -325,13 +329,18 @@ func (api *API) getTxResult(env *traceEnv, state *state.StateDB, index int, bloc
 			}
 			env.sMu.Lock()
 			m[keyStr] = wrappedProof
-			if sibling != nil {
-				env.DeletionProofs = append(env.DeletionProofs, sibling)
-			}
 			env.sMu.Unlock()
 		}
 	}
 
+	deletionProofs := tracer.DeletionProofs()
+	for _, bt := range deletionProofs {
+		key := crypto.Keccak256Hash(bt)
+		env.sMu.Lock()
+		env.deletionProofAgg[key] = bt
```

Comment:
`deletionProof` is not a full proof but just the bytes of a single node (the sibling of deleted leaf node) so each item is unique. From logtrace we may collect the same node so here we use a map for deduplication.

Reply:
Bytes of a node's value? Or its key? If we delete, reinsert, then delete again, this will work?

Reply:
It is in fact bytes of 'the whole node', i.e. trie can read these bytes as a valid node and add it to the (temporary) database.

Such an entry of node in database is immutable and not affected by insert / updating or deletion. It would just lost any reference from trie when being deleted. So adding any node into database would never break the trie, it just take no use in the worst case.